ansible/templates/usb/scripts/prompt-config.sh.j2

```diff
@@ -685,15 +698,16 @@ nicsup()
 		ifconfig $admin_iface_active inet $admin_ip netmask $admin_netmask up
 	fi
 
-	if [[ -n ${external_nic} ]]; then
-		if [ -n "$external_vlan_id" ]; then
-			vlan_opts="-v $external_vlan_id"
+	# create an interface for each nic tag with defined IP address
+	for ntag in "${!nictag_IPs[@]}"; do
+		if [ -n "${nictag_vlans[$ntag]}" ]; then
+			vlan_opts="-v ${nictag_vlans[$ntag]}"
 		fi
-
-		dladm create-vnic -l $external_iface $vlan_opts external0
-		ifconfig external0 plumb
-		ifconfig external0 inet $external_ip netmask $external_netmask up
-	fi
+		dladm create-vnic -l ${nictag_ifaces[$ntag]} $vlan_opts "${ntag}0"
```

Comment:
Seems like a defensive coding advice to me. It is very far from this joyent's script. But if you don't mind diverging more than necessary, why not.
BTW: this variable will never be empty and if so, It's a critical bug that needs to fail anyway.